src/ui/projectmanager/pm.go

```diff
@@ -26,7 +26,20 @@ type ProjectManager interface {
 	Exist(projectIDOrName interface{}) bool
 	GetRoles(username string, projectIDOrName interface{}) []int
 	// get all public project
-	GetPublic() []models.Project
+	GetPublic() []*models.Project
 	// get projects which the user is a member of
-	GetByMember(username string) []models.Project
+	GetByMember(username string) []*models.Project
+	Create(*models.Project) (int64, error)
+	Delete(projectIDOrName interface{}) error
+	Update(projectIDOrName interface{}, project *models.Project) error
+	// GetAll returns a project list and the total count according to
+	// the query conditions:
+	// owner: username of owner
+	// name: name of project
+	// public: public or not, can be "true", "false" or ""
+	// member: username of the member
+	// role: the role of member specified by member parameter
+	// page, size: pagination parameters
+	GetAll(owner, name, public, member string, role int, page,
```

Comment:
This long list of parms is a bit error prone as a public func
1, public should be bool
2,And why we need the role parm?  in this PR, I see all "role" are zero.  And since member and role are tied so probably put them in one struct.
3. page and size are tied so probably put them into one struct.  And I don't think the project service support pagination
4, as for the second return value, does it has to be returned in one call?
5, I see in the implementation you use "filter" for different functions, it's acceptable as it's internal, but why we need a "GetAll" public func?